hydrus/app.py

```diff
@@ -1,443 +1,444 @@
-    """Set the response headers."""
-    resp.status_code = status_code
-    for header in headers:
-        resp.headers[list(header.keys())[0]] = header[list(header.keys())[0]]
-    resp.headers['Content-type'] = ct
-    resp.headers['Link'] = '<' + get_hydrus_server_url() + \
-        get_api_name()+'/vocab>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"'
-    return resp
-
-
-def hydrafy(object_):
-    """Add hydra context to objects."""
-    object_["@context"] = "/"+get_api_name()+"/contexts/" + object_["@type"] + ".jsonld"
-    return object_
-
-
-def checkEndpoint(method, type_):
-    """Check if endpoint and method is supported in the API."""
-    for endpoint in get_doc().entrypoint.entrypoint.supportedProperty:
-        if type_ == endpoint.name:
-            for operation in endpoint.supportedOperation:
-                if operation.method == method:
-                    return True
-    return False
-
-
-def getType(class_type, method):
-    """Return the @type of object allowed for POST/PUT."""
-    for supportedOp in get_doc().parsed_classes[class_type]["class"].supportedOperation:
-        if supportedOp.method == method:
-            return supportedOp.expects.replace("vocab:", "")
-    # NOTE: Don't use split, if there are more than one substrings with 'vocab:' not everything will be returned.
-
-
-def checkClassOp(class_type, method):
-    """Check if the Class supports the operation."""
-    for supportedOp in get_doc().parsed_classes[class_type]["class"].supportedOperation:
-        if supportedOp.method == method:
-            return True
-    return False
-
-
-class Index(Resource):
-    """Class for the EntryPoint."""
-
-    def get(self):
-        """Return main entrypoint for the api."""
-        return set_response_headers(jsonify(get_doc().entrypoint.get()))
-
-
-class Vocab(Resource):
-    """Vocabulary for Hydra."""
-
-    def get(self):
-        """Return the main hydra vocab."""
-        return set_response_headers(jsonify(get_doc().generate()))
-
-
-class Entrypoint(Resource):
-    """Hydra EntryPoint."""
-
-    def get(self):
-        """Return application main Entrypoint."""
-        response = {"@context": get_doc().entrypoint.context.generate()}
-        return set_response_headers(jsonify(response))
-
-
-class Item(Resource):
-    """Handles all operations(GET, POST, PATCH, DELETE) on Items (item can be anything depending upon the vocabulary)."""
-
-    def get(self, id_, type_):
-        """GET object with id = id_ from the database."""
-        if get_authentication():
-            if request.authorization is None:
-                return failed_authentication()
-            else:
-                try:
-                    auth = check_authorization(request, get_session())
-                    if auth is False:
-                        return failed_authentication()
-                except Exception as e:
-                    status_code, message = e.get_HTTP()

```

Comment:
![sketch 43](https://user-images.githubusercontent.com/9114274/36605161-f05bcaec-18e5-11e8-8e4e-f341a2843535.png)

Reply:
Adapted to work with all python 3 versions

Reply:
This is an issue with Exception handling. Type annotations are fine. As you can see in the screenshot there are no type errors.
I will open one more PR to solve that exception handling issue.
Thanks for reviewing!